lang/syn/src/parser/context.rs

```diff
@@ -181,6 +218,21 @@ impl ParsedModule {
         })
     }
 
+    fn unsafe_struct_fields(&self) -> impl Iterator<Item = &syn::Field> {
```

Comment:
would it make sense to exclude accounts marked with `seeds` and `init` here? 

`seeds` already ensures it's the correct address

and `init` either creates a random new account or a PDA in which case `seeds` also makes sure it's the correct account

Reply:
I had this for [seeds](https://github.com/project-serum/anchor/blob/ff5d58413dc082232b413e9594a23cd082f22e7c/lang/syn/src/parser/context.rs#L58-L64) and[ init ](https://github.com/project-serum/anchor/pull/1452/commits/78f0cdf9fc4d972e14cc8f10baacc2e8aae42038 )at different commits but ended up removing it. I'll add it back if I'm not the only one that thought `seeds` and `init` is OK.

I also thought about owner checks added through the `account` attribute but I think that might be a step too far.

Reply:
One question is exactly what we want `/// CHECKED:` to mean.

If it is that *both* the owner and the type are verified, then skipping `seeds` and `init` isn't necessarily safe. Because one can use these keywords with `UncheckedAccount` or `AccountInfo` and subsequently forget the check (or in the case of init, fail to write the header).